xblock/fields.py

```diff
@@ -720,7 +712,6 @@ class Float(JSONField):
     something for which float(value) does not throw an error.
 
     """
-    MUTABLE = False
```

Comment:
Hmm, looking at this more closely, there might be a memory usage implication for this change.  It may produce more copies of the defaults than we want for certain objects.   Perhaps this is something we need to keep after all.

My thought is this: If whenever we ask for the default value of certain fields, it makes a new copy of the default value via `deepcopy` then we will increase the memory usage of the application each time a new default is loaded.  Given that these are fields in xblocks, that might happen quite a bit, so I wonder if it actually makes more sense to not make this change and instead decide that we're happy with the code as is.  @salman2013 @farhan what do you guys think?

Reply:
@feanil I think your point of increasing memory usage makes sense and we should not make this change, actually when i removed it the perception in my mind was the value MUTABLE  is true by default 

>  MUTABLE = True
    _default = None
    # Indicates if a field's None value should be sent to the XML representation.
    none_to_xml = False
@@ -334,10 +333,7 @@ def __init__(self, help=None, default=UNSET, scope=Scope.content,  # pylint:disa
    @property
    def default(self):
        """Returns the static value that this defaults to."""
        if self.MUTABLE:
            return copy.deepcopy(self._default)

and its already making a deep copy and maybe it would be for some reason but yeah your point is valid it would increase app memory size whenever it is used.

Reply:
Ok, let's abandon this PR and close the related issue with a link to this PR and a summary of the reason why we  did not pursue it.